pkg/kwokctl/cmd/create/cluster/cluster.go

```diff
@@ -127,14 +128,35 @@ func runE(ctx context.Context, flags *flagpole) error {
 		defer cancel()
 	}
 
-	buildRuntime, ok := runtime.DefaultRegistry.Get(flags.Options.Runtime)
-	if !ok {
-		return fmt.Errorf("runtime %q not found", flags.Options.Runtime)
+	var runtimeCandidate []string
+	if flags.Options.Runtime != "" {
+		runtimeCandidate = []string{flags.Options.Runtime}
+	} else {
+		runtimeCandidate = []string{
+			consts.RuntimeTypeDocker,
+			consts.RuntimeTypeNerdctl,
+			consts.RuntimeTypeBinary,
+			consts.RuntimeTypeKind,
+		}
 	}
 
-	rt, err := buildRuntime(name, workdir)
+	var rt runtime.Runtime
+	var err error
+	var errs []error
+	for _, r := range runtimeCandidate {
+		buildRuntime, ok := runtime.DefaultRegistry.Get(r)
+		if !ok {
+			continue
+		}
+		rt, err = buildRuntime(name, workdir)
```

Comment:
We need to add a new method to the Cluster interface to check if it can run

Reply:
the `buildRuntime` has returned an error previously so I used it to do the basic runtime checks.

we can discuss a new method if necessary

Reply:
Yeah, add a new method, not in NewCluster

Reply:
may I ask why preferring a new method? what's the point of returning err for `NewCluster` if we always return nil?

Reply:
Runtime detection is only required when creating a cluster, and is not required and cannot be switched in other.

Reply:
nope, you need to know which runtime to use when even updating, deleting

Reply:
Cluster creation will record the runtime, and the second time will use the previous Runtime, instead of re-detecting it.